### PR TITLE
Add configs for other versioned upgrade tests

### PR DESCRIPTION
**Pull request by ihmccreery**
For 1.0-1.2, 1.1-1.3, and 1.2-1.3, and remove old upgrade jobs.

### CODE REVIEW

File: jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e.yaml [hunk 1]
```diff
@@ -1034,18 +1034,56 @@
...
+                # XXX This is a hack to run only the tests we care about, without importing all of
+                # the skip list vars from the v1.0 e2e.sh.
+                export GINKGO_TEST_ARGS="--ginkgo.skip=Skipped|Restart\sshould\srestart\sall\snodes|Example|Reboot|ServiceLoadBalancer|DaemonRestart\sController\sManager|Daemon\sset\sshould\srun\sand\sstop\scomplex\sdaemon|Resource\susage\sof\ssystem\scontainers|allows\sscheduling\sof\spods\son\sa\sminion\safter\sit\srejoins\sthe\scluster"
+        - 'kubernetes-e2e-{provider}-{version-old}-{version-new}':
             version-old: '1.1'
             version-new: '1.2'
             version-infix: '1-1-1-2'
             version-env: |
-                # In v1.1, NUM_MINIONS defaults to 2, so we have to override to 3 here.  NUM_MINIONS
+                # In v1.1-, NUM_MINIONS defaults to 2, so we have to override to 3 here.  NUM_MINIONS
```
Comment (spxtr) [inline jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e.yaml]: Why the dash after `v1.1`?

Comment (ihmccreery) [inline jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e.yaml]: Er, v1.1 and below (as opposed to v1.2+).  If that's confusing, which I guess it is, would "and below" be clearer?

**Comment by ihmccreery** [comment]
@spxtr PTAL.

**Comment by spxtr** [comment]
It looks fine to me, but I only glanced over the diff. Will there be jobs to delete? I think we can actually turn on `--delete-old` on post-commit JJB.

**Comment by ihmccreery** [comment]
There will be lots of old jobs to delete; you can either grant me RW temporarily and I can do it myself, or I can give you the list and you can.

File: jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e.yaml (hunk 1 above)
Comment (spxtr) [inline jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e.yaml]: Oh! I see. Yes, that makes more sense to me. Thanks.

**Pull request merged**